appcues/src/main/java/com/appcues/ui/composables/StepComposition.kt

```diff
@@ -4,69 +4,95 @@ import androidx.compose.foundation.layout.Box
-            .verticalScroll(rememberScrollState())
-            // if we have contentPadding to apply from the WrapContent trait then we apply here
-            .then(if (contentPadding != null) Modifier.padding(contentPadding) else Modifier)
-            .padding(paddingValues = stepDecoratingPadding.paddingValues.value)
-            .testTag("page_$index")
+internal fun Step.ComposeStep(modifier: Modifier = Modifier, wrapperInsets: PaddingValues, parent: BoxScope) {
+    CompositionLocalProvider(
+        LocalAppcuesActions provides actions,
+        LocalExperienceStepFormStateDelegate provides formState
     ) {
-        content.Compose()
+        // used to get the padding values from step decorating trait and apply to the Column
+        val density = LocalDensity.current
+        val stickyContentPadding = remember(this) { StickyContentPadding(density) }
+
+        ApplyUnderlayStepTraits(parent, wrapperInsets, stickyContentPadding)
+
+        ComposeStepContent(modifier, wrapperInsets, stickyContentPadding)
+
+        ComposeStickyContent(parent, wrapperInsets, stickyContentPadding)
+
+        ApplyOverlayStepTraits(parent, wrapperInsets, stickyContentPadding)
     }
 }
 
 @Composable
-internal fun Step.ApplyUnderlayStepTraits(boxScope: BoxScope, stepDecoratingPadding: StepDecoratingPadding) {
+private fun Step.ApplyUnderlayStepTraits(
+    boxScope: BoxScope,
+    wrapperInsets: PaddingValues,
+    stickyContentPadding: StickyContentPadding
+) {
     stepDecoratingTraits
         .filter { it.stepComposeOrder == StepDecoratingType.UNDERLAY }
-        .forEach { it.run { boxScope.DecorateStep(stepDecoratingPadding) } }
+        .forEach { it.run { boxScope.DecorateStep(wrapperInsets, stickyContentPadding) } }
 }
 
 @Composable
-internal fun Step.ApplyOverlayStepTraits(boxScope: BoxScope, stepDecoratingPadding: StepDecoratingPadding) {
+private fun Step.ApplyOverlayStepTraits(
+    boxScope: BoxScope,
+    wrapperInsets: PaddingValues,
+    stickyContentPadding: StickyContentPadding
+) {
     stepDecoratingTraits
         .filter { it.stepComposeOrder == StepDecoratingType.OVERLAY }
-        .forEach { it.run { boxScope.DecorateStep(stepDecoratingPadding) } }
+        .forEach { it.run { boxScope.DecorateStep(wrapperInsets, stickyContentPadding) } }
+}
+
+@Composable
+private fun Step.ComposeStepContent(
+    modifier: Modifier,
+    wrapperInsets: PaddingValues,
+    stickyContentPadding: StickyContentPadding
+) {
+    Column(
+        horizontalAlignment = Alignment.CenterHorizontally,
+        modifier = modifier
+            .padding(wrapperInsets)
+            .padding(stickyContentPadding.paddingValues.value)
+    ) { content.Compose() }
 }
 
 @Composable
-internal fun Step.ComposeStickyContent(boxScope: BoxScope, stepDecoratingPadding: StepDecoratingPadding) {
+private fun Step.ComposeStickyContent(
+    boxScope: BoxScope,
+    wrapperInsets: PaddingValues,
+    stickyContentPadding: StickyContentPadding
+) {
```

Comment:
Wondering if we'll have a complicated merge here if we make changes in how sticky works in the 1.3.x branch for the recent bug issue (not respecting modal padding) and then need to merge here, but we'll figure it out.

Reply:
not sure how thats going to work either. we will see